lib/faraday/request/multipart.rb

```diff
@@ -61,6 +61,21 @@ def create_multipart(env, params)
         body
       end
 
+      def part(boundary, key, value)
+        if json?(value)
+          header = { 'Content-Type' => 'application/json' }
+          Faraday::Parts::Part.new(boundary, key, value, header)
+        else
+          Faraday::Parts::Part.new(boundary, key, value)
+        end
+      end
+
+      def json?(value)
+        JSON.parse(value)
```

Comment:
:thinking: I'm wondering if it may make sense to test if `value` is a string and maybe starts with `{` (maybe with some optional preceding whitespace?) ?

This way we will fail earlier if it is not json, and would avoid having to raise/collect stacktrace/rescue exception

WDYT?